modules/virt-cluster-resource-requirements.adoc

```diff
@@ -0,0 +1,37 @@
+// Module included in the following assemblies:
+//
+// * virt/install/preparing-cluster-for-virt.adoc
+
+[id="virt-cluster-resource-requirements{context}"]
+= Additional hardware requirements for {VirtProductName}
+
+
+{VirtProductName} is an add-on to {product-title} and imposes additional overhead that you need to take into account when planning a cluster. Each cluster machine must meet the following hardware requirements in addition to the {product-title} requirements.
+
+.Memory
+
+----
+Memory overhead per virtual machine = (1.002 * requested memory) + 146 MiB  \
+                + (8 * number of vCPUs) + (16 MiB * number of graphics devices)
+----
+
+Add 1 GiB to the above result if your environment includes a Single Root I/O Virtualization (SR-IOV) network device or a Graphics Processing Unit (GPU).
+
+----
+Cluster memory overhead = 2179 MiB + (366 MiB * number of nodes) \
+               + memory overhead per virtual machine
+----
+
+
+.CPU
+
+CPU overhead per virtual machine is subjective. If dedicated CPUs are requested, there is a 1:1 impact on cluster CPU requirement. Otherwise there is no rule of thumb about how many CPUs a virtual machine would require.
+
+
+.Storage
+
+----
+Storage overhead per virtual machine = 11.12 GB or 10.36 GiB
```

Comment:
After re-reading this line I just realized it is incorrect. The storage overhead numbers listed here are the on-disk impact of the sum total of all CNV containers. In other words this is the storage impact of the CNV itself. This impact is per-node as OpenShift must download a copy of each container locally.

We had originally compiled a list of containers per infra node or worker node, only to realize after the fact that OLM will download everything in the catalogue, regardless of whether a particular node will ever run it or not.

------------

Storage requirements per Virtual Machine are entirely independent of this. Any impact a VM has will be due to specific request for resource allocation within the VM. Whether that is ephemeral on the node or storage resources hosted elsewhere in the cluster.

CNV does not currently allocate any additional ephemeral storage for the running container itself.